onnxruntime/core/providers/cpu/nn/pool.cc

```diff
@@ -216,9 +218,15 @@ Status Pool<float, AveragePool>::Compute(OpKernelContext* context) const {
 
 template <>
 Status Pool<float, MaxPool<8 /*VERSION*/>>::Compute(OpKernelContext* context) const {
-  // Use MLAS pooling if the index output tensor is not used.
-  if (OpKernel::Node().OutputDefs().size() == 1) {
-    return PoolBase::Compute(context, MlasMaximumPooling);
+  // Use MLAS pooling if the index output tensor is not used
+  // and also if dilation is not required
+
+  bool need_dilation = false;
+  for (auto n : dilations_)
+    need_dilation |= n > 1;
+
+  if (OpKernel::Node().OutputDefs().size() == 1 && !need_dilation) {
```

Comment:
Nothing seems to be happening in this 'if' block.

Reply:
Good catch -- uncommented the line in that block. If dilations is not required, we can use the more optimized MLAS execution block instead.